tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/LiferayWorkspaceProject.java

```diff
@@ -97,7 +99,19 @@ public LiferayWorkspaceProject(IProject project) {
 
 	@Override
 	public String getProperty(String key, String defaultValue) {
-		return null;
+		Properties properties = new Properties();
+
+		IPath projectLocation = getProject().getLocation();
+
+		IPath bladeDir = projectLocation.append(".blade");
+
+		IPath bladeSettingsPath = bladeDir.append("settings.properties");
+
+		properties.putAll(PropertiesUtil.loadProperties(bladeSettingsPath));
+
+		properties.putAll(loadExtraProperties());
```

Comment:
This should not happen here.  This should happen with this workspace project is created and all the properties should be read.   So each workspaceProject would have a properties object and could be read from this getProperty() method.  There is no need for a protected `loadExtraProperties` method.